src/program.cpp

```diff
@@ -205,8 +208,70 @@ target_assignments program::get_target_assignments(const std::vector<target>& ta
 
 bool program::is_compiled() const { return not this->impl->target_name.empty(); }
 
+void program::compile(const std::vector<std::string>& targets,
+                      std::unordered_map<std::string, compile_options> compile_opt_map)
```

Comment:
This can probably be `std::vector<compile_options>`

Reply:
That would require knowing the index of the target each time we reference `compile_options`.

Reply:
If the `roots` is `std::unordered_multimap<std::size_t, module_ref>` then you should have the index when you iterate the roots.

Reply:
Done.